cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -447,21 +531,12 @@ func (scaleSet *ScaleSet) DeleteInstances(instances []*azureRef, hasUnregistered
 
 	ctx, cancel := getContextWithTimeout(vmssContextTimeout)
 	defer cancel()
-	resourceGroup := scaleSet.manager.config.ResourceGroup
 
-	scaleSet.instanceMutex.Lock()
-	klog.V(3).Infof("Calling virtualMachineScaleSetsClient.DeleteInstancesAsync(%v), force delete set to %v", requiredIds.InstanceIds, scaleSet.enableForceDelete)
-	future, rerr := scaleSet.manager.azClient.virtualMachineScaleSetsClient.DeleteInstancesAsync(ctx, resourceGroup, commonAsg.Id(), *requiredIds, scaleSet.enableForceDelete)
+	future, rerr := scaleSet.deleteInstances(ctx, requiredIds, commonAsg.Id())
 
-	if scaleSet.enableForceDelete && isOperationNotAllowed(rerr) {
```

Comment:
Not present on older versions due to #6447. But seemingly justified(?) due to fork code differences, see azure_force_delete_scale_set.go for after-fork location.

This feature itself and the after-fork location is already in the fork from past versions, but #6447 upstream it in master.